src/Mailbox.php

```diff
@@ -61,6 +71,18 @@ public function getMessages(SearchExpression $search = null)
         $query = ($search ? (string) $search : 'ALL');
 
         $messageNumbers = imap_search($this->connection->getResource(), $query, \SE_UID);
+
+        $lastError = imap_last_error();
+        if ($lastError !== false) {
+            if (1 === preg_match('/Unknown search criterion: ([A-Z]+)/', $lastError, $matches)) {
+                // drop php notices.
+                imap_errors();
+                throw new ServerDisallowCriterionException($matches[1]);
+            } else {
+                throw new Exception($lastError);
+            }
+        }
```

Comment:
Implemented in https://github.com/ddeboer/imap/pull/198/files#diff-fd5d8baa93f5124ecb7ec0f13038e744R119 thank you